### DISCUSSION TITLE
useArrayField fill with data from remote

### DISCUSSION THREAD

**Discussion by popeating** [react-hook-form/react-hook-form#9304] [Q&A] [answered]
I have a compoment as page to set options, one of the option is a field array

i load the option from remote and set them to the fieds with a useEffect:
```
  useEffect(() => {
    var options = JSON.parse(data.value).options;
    if (options) {
      setValue('closing_days', options.closing_days);
      setValue('max_days', options.max_days);
      setValue('require_confirm', options.require_confirm);
      setValue('time_span', options.time_span);
      setValue('maxPax', options.maxPax);
      options.times.map((time, i) => {
        console.log(time, i, 'slots');
        update(i, { value: time.value });
        setValue(`times.${i}.value`, time.value);
      });
    }

```

The array field is created as follow:
```
 <input
                    key={field.id}
                    {...register(`times.${index}.value`)}
                  />
```
form and field array are instantiated as follow:
```
  const {
    register,
    handleSubmit,
    setValue,
    control,
    formState: { errors },
  } = useForm();
  const {
    fields,
    append,
    prepend,
    remove,
    swap,
    move,
    insert,
    update,
    replace,
  } = useFieldArray({
    control,
    name: 'times', // unique name for your Field Array
  });
```

First time i load the component, the fieldArray shows the correct number of fields and they are also correctly filled with the values coming from data, but if i move to another componet and get back, the fieldArray is still present with  the correct number of fields, but only the last one got a value (the other ones are empty), all the regular fields are correctly set, the console log in the map function prints out all and the correct values.

Before coding my own array of fields, anyone got some suggestion?

**Comment by bluebill1049** [comment, marked as answer]
Have you tried with `reset` or `replace`?

**Comment by popeating** [comment]
Thank you! repalce did the job, i was stuck using setValue
